src/navigation/actions.js

```diff
@@ -34,6 +34,29 @@ import { PREFERENCE_KEYS } from '../database/utilities/constants';
  *
  */
 
+export const createCustomerRequisition = ({
+  currentUser,
+  ...requisitionParameters
+}) => dispatch => {
+  // Months lead time has an effect on daysToSupply for a requisition.
+
+  // Create the requisition. If a program was supplied, add items from that
+  // program, otherwise just navigate to it.
+  let requisition;
+  UIDatabase.write(() => {
+    requisition = createRecord(
+      UIDatabase,
+      'CustomerRequisition',
+      currentUser,
+      requisitionParameters
+    );
+
+    if (requisition.program) requisition.addItemsFromProgram(UIDatabase);
```

Comment:
When adding column/new fields, I think this is where more changes will be needed: Possibly an `addCustomerRequisitionItem` or something

Reply:
I uhh.. didn't think about that. So you're saying a customer requisition which comes through sync should not have editable columns for the form entry? Just the `supply this invoice` field?

Reply:
Yup, they should not be editable if the customer requisition come from a mSupply store. mSupply desktop does this by checking `[requisition]linked_requisition_id`. If it is blank we infer this response requisition isn't linked to a request, so must be a manual one.

Unfortunately, mobile doesn't have this field. If we did add it, for historic requisitions we'd have to resync for them to work (thankfully just the requisition records, not their lines). Furthermore we'd have to either make a placeholder requisition in realm to link to, or just save it as a string (optional/default `''`) field. The latter would be adequate, at least till we have multi store mobile!

Reply:
We can defer addressing this to a new issue

Reply:
Cool - I'll sort this out in https://github.com/openmsupply/mobile/issues/3143 - Although I think we're disagreeing on the `linked_requisition_id` - the addition of having all mobile requisitions resync when upgrading to 6.0+ is another decent downside?

Reply:
Significant downside yes, but can discuss in that issue you've highlighted 🙂